pkg/credentialprovider/plugin/plugin.go

```diff
@@ -532,42 +606,31 @@ func (p *pluginProvider) getCachedCredentials(image, serviceAccountCacheKey stri
 	}
 	p.Unlock()
 
-	cacheKey, err := generateCacheKey(image, serviceAccountCacheKey)
-	if err != nil {
-		return nil, false, fmt.Errorf("error generating cache key: %w", err)
-	}
-
-	obj, found, err := p.cache.GetByKey(cacheKey)
-	if err != nil {
-		return nil, false, err
-	}
-
-	if found {
-		return obj.(*cacheEntry).credentials, true, nil
-	}
-
-	registry := parseRegistry(image)
-
-	cacheKey, err = generateCacheKey(registry, serviceAccountCacheKey)
-	if err != nil {
-		return nil, false, fmt.Errorf("error generating cache key: %w", err)
+	cacheKeyCandidates := []string{
+		image,
+		parseRegistry(image),
+		globalCacheKey,
```

Comment:
consider whether to fix the collision risk between the magic "global" constant and an image or registry using the literal exact string "global", either by switching `globalCacheKey` to something that would be invalid as an image or registry, or by tag-prefixing the cache key types